app/src/main/java/com/jetpack/compose/learning/viewpager/AddRemovePagerActivity.kt

```diff
@@ -0,0 +1,276 @@
+                        dropDownClick(it)
+                    }
+                    SingleLineTextWithIcon(Icons.Filled.Delete, OptionType.DELETE.value) {
+                        dropDownClick(it)
+                    }
+                }
+            },
+            navigationIcon = {
+                IconButton(onClick = { onBackPressed() }) {
+                    Icon(Icons.Filled.ArrowBack, contentDescription = null)
+                }
+            }
+        )
+    }
+
+    @Composable
+    fun AddOrRemovePageDialog(
+        optionType: String,
+        totalListCount: Int,
+        dismissDialog: () -> Unit,
+        addRemoveButtonClick: (Int) -> Unit
+    ) {
+        val positionValueTextField by viewModel.positionValueTextField.collectAsState()
+        val isShowPositionError by viewModel.isShowPositionError.collectAsState()
+
+        Dialog(
+            onDismissRequest = {
+                dismissDialog()
+            },
+        ) {
+            Card(
+                modifier = Modifier
+                    .width(320.dp)
+                    .height(190.dp)
+                    .clip(RoundedCornerShape(8.dp))
+                    .background(Color.White)
+            ) {
+                Column {
+                    Text(
+                        "Enter position to $optionType page",
+                        modifier = Modifier.padding(15.dp)
+                    )
+                    OutlinedTextField(
+                        value = positionValueTextField,
+                        onValueChange = {
+                            viewModel.isShowPositionError.value = false
+                            if (it.isDigitsOnly())
+                                viewModel.positionValueTextField.value = it
+                        },
+                        modifier = Modifier.padding(start = 15.dp),
+                        keyboardOptions = KeyboardOptions.Default.copy(
+                            keyboardType = KeyboardType.Number,
+                            imeAction = ImeAction.Done
+                        ),
+                        isError = isShowPositionError,
+                    )
+                    if (isShowPositionError) {
+                        Text(
+                            stringResource(R.string.text_invalid_position),
+                            modifier = Modifier.padding(start = 15.dp),
+                        )
+                    }
+                    Row(
+                        horizontalArrangement = Arrangement.End,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(12.dp)
+                    ) {
+                        TextButton(onClick = {
+                            dismissDialog()
+                        }) {
+                            Text(stringResource(R.string.btn_cancel))
+                        }
+                        TextButton(onClick = {
+                            when {
```

Comment:
Optimize this when condition because for both the condition we are invoking same function -> addRemoveButtonClick(positionValueTextField.toInt())